mlflow/tensorflow.py

```diff
@@ -989,110 +985,118 @@ def _log_early_stop_callback_metrics(callback, history, metrics_logger):
+    def fit_generator(original, self, *args, **kwargs):
         """
         NOTE: `fit_generator()` is deprecated in TF >= 2.1.0 and simply wraps `fit()`.
         To avoid unintentional creation of nested MLflow runs caused by a patched
         `fit_generator()` method calling a patched `fit()` method, we only patch
         `fit_generator()` in TF < 2.1.0.
         """
-        with _manage_active_run() as run:
-            original = gorilla.get_original_attribute(tensorflow.keras.Model, "fit_generator")
-
-            unlogged_params = ["self", "generator", "callbacks", "validation_data", "verbose"]
-
-            log_fn_args_as_params(original, args, kwargs, unlogged_params)
-
-            run_id = run.info.run_id
-            with batch_metrics_logger(run_id) as metrics_logger:
-                # Checking if the 'callback' argument of fit() is set
-                if len(args) >= 5:
-                    tmp_list = list(args)
-                    tmp_list[4], log_dir = _setup_callbacks(tmp_list[4], log_models, metrics_logger)
-                    args = tuple(tmp_list)
-                elif kwargs.get("callbacks"):
-                    kwargs["callbacks"], log_dir = _setup_callbacks(
-                        kwargs["callbacks"], log_models, metrics_logger
-                    )
-                else:
-                    kwargs["callbacks"], log_dir = _setup_callbacks([], log_models, metrics_logger)
-                result = original(self, *args, **kwargs)
+        active_run = mlflow.active_run()
+        if MLFLOW_AUTOLOGGING in active_run.data.tags:
+            global _AUTOLOG_RUN_ID
+            _AUTOLOG_RUN_ID = active_run.info.run_id
 
-            _flush_queue()
-            _log_artifacts_with_warning(
-                local_dir=log_dir.location, artifact_path="tensorboard_logs"
-            )
-            if log_dir.is_temp:
-                shutil.rmtree(log_dir.location)
+        unlogged_params = ["self", "generator", "callbacks", "validation_data", "verbose"]
+
+        log_fn_args_as_params(original, args, kwargs, unlogged_params)
+
+        run_id = active_run.info.run_id
+
+        with batch_metrics_logger(run_id) as metrics_logger:
+            # Checking if the 'callback' argument of fit() is set
+            if len(args) >= 5:
+                tmp_list = list(args)
+                tmp_list[4], log_dir = _setup_callbacks(tmp_list[4], log_models, metrics_logger)
+                args = tuple(tmp_list)
+            elif kwargs.get("callbacks"):
+                kwargs["callbacks"], log_dir = _setup_callbacks(
+                    kwargs["callbacks"], log_models, metrics_logger
+                )
+            else:
+                kwargs["callbacks"], log_dir = _setup_callbacks([], log_models, metrics_logger)
+            result = original(self, *args, **kwargs)
+
+        _flush_queue()
+        _log_artifacts_with_warning(local_dir=log_dir.location, artifact_path="tensorboard_logs")
+        if log_dir.is_temp:
```

Comment:
Same as https://github.com/mlflow/mlflow/pull/3861/files#r545584715 - can we leverage `mlflow.utils.autologging_utils.PatchFunction.on_exception`?

Reply:
Done!